client/src/pages/CalendarForm.tsx

```diff
@@ -84,6 +87,22 @@ export function CalendarForm() {
     }
   }
 
+  async function handleDelete() {
+    try {
+      if (!calendarId) throw new Error("shouldn't happen");
+      const deletedCal = await deleteCal(calendarId);
+      alert(`Deleted ${deletedCal.name}`);
+      setName(undefined);
```

Comment:
Since you are navigating away, I'm not sure you need to reset these state values. It depends, though, on how the CalendarForm is rendered.